cost/azure/instances_log_analitics_utilization/azure_instance_log_analytics_utilization.pt

```diff
@@ -318,87 +342,125 @@ script "js_oms_resources", type: "javascript" do
+        }else if (counterName === "Available MBytes Memory") {
+          // Linux counter is available/free, not used
+          var instanceDetails = _.where(ds_azure_vm_sizes, {"name": instance["vmSize"]})
+          var totalMem = instanceDetails[0].MemoryGB * 1024
+          mem_maximum = parseFloat(((totalMem - metric.min) / totalMem) * 100).toFixed(2)
+          mem_average = parseFloat(((totalMem - metric.avg) / totalMem) * 100).toFixed(2)
+        }else if (counterName === "% Committed Bytes In Use") {
+          mem_maximum = parseFloat(metric.max).toFixed(2)
+          mem_average = parseFloat(metric.avg).toFixed(2)
+        }
+      })
+
+      result.push({
+        subscriptionId: instance["subscriptionId"]
+        subscriptionName:instance["subscriptionName"]
+        id: instance["id"],
+        name: instance["name"],
+        resourceGroup: instance["resourceGroup"],
+        location: instance["location"],
+        vmSize: instance["vmSize"],
+        osType: instance["osType"],
+        logAgent: instance["logAgent"],
+        workspaceId: instance["workspaceId"],
+        diagAgent: instance["diagAgent"],
+        storageAccount: instance["storageAccount"],
+        tags: instance["tags"],
+        availabilitySet: instance["availabilitySet"]
+        cpu_maximum: cpu_maximum,
+        cpu_average: cpu_average,
+        mem_maximum: mem_maximum,
+        mem_average: mem_average
+      })
+    }
+  })
+EOS
+end
 
-        result.push({
-          id: instance["id"],
-          name: instance["name"],
-          resourceGroup: instance["resourceGroup"],
-          location: instance["location"],
-          vmSize: instance["vmSize"],
-          osType: instance["osType"],
-          logAgent: instance["logAgent"],
-          workspaceId: instance["workspaceId"],
-          diagAgent: instance["diagAgent"],
-          storageAccount: instance["storageAccount"],
-          tags: instance["tags"],
-          availabilitySet: instance["availabilitySet"]
-          cpu_maximum: cpu_maximum,
-          cpu_average: cpu_average,
-          mem_maximum: mem_maximum,
-          mem_average: mem_average
-        })
-      }
-    })
-  EOS
+script "js_inefficient_utilization_metrics", type: "javascript" do
+  result "inefficient_utilization_results"
+  parameters "ds_azure_instance_size_map", "ds_merged_metrics", "param_avg_used_memory", "param_avg_cpu"
+  code <<-EOS
+  var inefficient_utilization_results = []
+  for ( i = 0; i < ds_merged_metrics.length; i++){
+    var instance = ds_merged_metrics[i]
+    var cpu_average = instance.cpu_average
+    var mem_average = instance.mem_average
+    var next_vm_size = ds_azure_instance_size_map[instance.vmSize].down
+    if(next_vm_size == null && typeof next_vm_size == "undefined" ){
+      next_vm_size="N/A";
+    }
+    if (cpu_average < param_avg_cpu || mem_average < param_avg_used_memory){
+      inefficient_utilization_results.push({
```

Comment:
sort by the subscriptonName then location.

Reply:
Can you please give a little more clarity on this one? What should be the sorting based on?

Reply:
I would like to see the sorting by Subscription Name and the location.  There could be multiple locations in the same subscription. Sort by subscription name first, then location.

Reply:
I have added the code for same. It is now sorting by subscription and location.